src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/PropertyBasedEmbeddingServiceConfig.java

```diff
@@ -1,40 +1,94 @@
 package io.stargate.sgv2.jsonapi.service.embedding.configuration;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import io.smallrye.config.ConfigMapping;
 import io.smallrye.config.WithConverter;
 import io.smallrye.config.WithDefault;
 import jakarta.annotation.Nullable;
-import java.net.URL;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import org.eclipse.microprofile.config.spi.Converter;
 
-@ConfigMapping(prefix = "stargate.jsonapi.embedding.service")
+@ConfigMapping(prefix = "jsonapi.vector.provider")
 public interface PropertyBasedEmbeddingServiceConfig {
 
   @Nullable
-  ServiceConfig openai();
+  VectorProviderConfig openai();
 
   @Nullable
-  ServiceConfig huggingface();
+  VectorProviderConfig huggingface();
 
   @Nullable
-  ServiceConfig vertexai();
+  VectorProviderConfig vertexai();
 
   @Nullable
-  ServiceConfig cohere();
+  VectorProviderConfig cohere();
 
   @Nullable
-  ServiceConfig nvidia();
+  VectorProviderConfig nvidia();
 
   @Nullable
   CustomConfig custom();
 
-  public interface ServiceConfig {
+  interface VectorProviderConfig {
+    @JsonProperty
     boolean enabled();
 
-    URL url();
+    @JsonProperty
+    String url();
 
+    @JsonProperty
     String apiKey();
+
+    @JsonProperty
+    List<String> supportedAuthentication();
+
+    @Nullable
+    @JsonProperty
+    List<ParameterConfig> parameters();
+
+    @Nullable
+    @JsonProperty
```

Comment:
What is the difference between `parameters` and `properties`? I think Javadocs would make sense here to explain.

Reply:
I think `parameters` are used by the user to construct the URL or to customize the model. (external) `properties` are used by us to contact the real model. (internal) The example below gives the full configuration, you could see the difference that `parameters` are for the user to customize their model and `properties` are for us to actually use the model.

```
{
    "status": {
        "vectorProviders": {
            "vertexai": {
                "url": "https://us-central1-aiplatform.googleapis.com/v1/projects/{PROJECT_ID}/locations/us-central1/publishers/google/models/{MODEL}:predict",
                "enabled": true,
                "properties": {
                    "max-retries": "3",
                    "retry-delay-ms": "100",
                    "timeout-ms": "1000"
                },
                "parameters": [
                    {
                        "name": "PROJECT_ID",
                        "type": "STRING",
                        "defaultValue": null,
                        "help": null,
                        "required": true
                    }
                ],
                "supportedAuthentication": [
                    "HEADER",
                    "SHARED_SECRETS"
                ],
                "models": [
                    {
                        "name": "textembedding-gecko@003",
                        "properties": {
                            "max-tokens": "3072"
                        },
                        "parameters": [
                            {
                                "name": "autoTruncate",
                                "type": "BOOLEAN",
                                "defaultValue": "true",
                                "help": null,
                                "required": false
                            }
                        ]
                    }
                ]
            }
        }
    }
}
```